test/integration/visitor_redirected_to_login_when_clicks_checkout_test.rb

```diff
@@ -0,0 +1,26 @@
+require 'test_helper'
+
+class VisitorRedirectedToLoginWhenClicksCheckoutTest < ActionDispatch::IntegrationTest
+  test 'user that has not logged in directed to login when click checkout' do
+    create_shop
+    create_user
+    create_cart_for_visitor
+
+    visit '/cart'
+    click_button "Place Order"
+
+    assert_equal login_path, current_path
+    assert page.has_content?("Please login before placing order.")
+
+    login_user
+    visit '/cart'
+
+    assert page.has_content?("Cart (1)")
+
+    click_button "Place Order"
+
+    assert_equal '/orders', current_path
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.